tests/test_python_engine.py

```diff
@@ -389,6 +392,338 @@ def test_custom_functions(self):
+            ip_constructor = type(min_ip_address)
+            max_int = 2**32 - 1 if type(min_ip_address) == ipaddress.IPv4Address else 2**128 - 1
+
+            min_ip_address_int = int(min_ip_address)
+            max_ip_address_int = int(max_ip_address)
+
+            # Generate a random IP address.
+            if random.random() < 0.5:
+                random_ip_address = ip_constructor(random.randint(0, min_ip_address_int))
+            else:
+                random_ip_address = ip_constructor(random.randint(max_ip_address_int, max_int))
+
+            # Check if the random IP address is within the provided range.
+            while random_ip_address >= min_ip_address and random_ip_address <= max_ip_address:
+                # Generate another random IP address.
+                random_ip_address = ip_constructor(random.randint(0, min_ip_address_int))
+                if random_ip_address >= min_ip_address and random_ip_address <= max_ip_address:
+                    random_ip_address = ip_constructor(random.randint(max_ip_address_int, max_int))
+                max_tries -= 1
+                if max_tries == 0:
+                    return random_ip_address
+
+            return random_ip_address
+
+        config = {"flatten": True}
+        events = [
+            Event.from_data(d)
+            for d in [
+                {
+                    "event_type": "process",
+                    "process_name": "malicious.exe",
+                    "unique_pid": "host1-1",
+                    "timestamp": 116444736000000000,
+                    "source": {"ip": "2001:0db8:0000:0000:0000:0000:0000:0001"},
+                },
+                {
+                    "event_type": "process",
+                    "process_name": "missing.exe",
+                    "unique_pid": "host1-1",
+                    "timestamp": 116444738000000000,
+                    "source": {"ip": "2001:db8::1"},
+                },
+                {
+                    "event_type": "file",
+                    "file_name": "suspicious.txt",
+                    "unique_pid": "host1-1",
+                    "timestamp": 116444740000000000,
+                    "source": {"ip": "fe80::1"},
+                },
+            ]
+        ]
+
+        # Should return no results none of the CIDR ranges contain those IPs
+        query = """
+        sequence by unique_pid with maxspan=7m
+            [ process where cidrMatch(source.ip, "10.0.0.1/8") ]
+            [ process where cidrMatch(source.ip, "10.0.0.1/16") ]
+            [ file where cidrMatch(source.ip, "10.0.0.1/32", "2001:0db8::/32") ]
+        """
+        with elasticsearch_syntax:
+            parsed_query = parse_query(query)
+
+        output = self.get_output(queries=[parsed_query], config=config, events=events)
+
+        self.assertEqual(len(output), 0, "Missing or extra results")
+
+        # Should return results since the cidr ranges match the addresses in the events
+        query = """
```

Comment:
Can you add a test that searches the known `events` where the shorthand is explicitly defined in the query (to validate expansion)